### PR TITLE
feat(android-setup): Add port forwarding steps and configuration

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Description of changes
When we put the dialogs together, we realized that we were never configuring the port forwarding. This PR adds 2 new steps--one for when we're configuring the port forwarding, and one for when we ran into a problem and need to have the user take some manual action.

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/electron/platform/android/setup/steps/configuring-port-forwarding.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AndroidSetupStepConfig } from 'electron/platform/android/setup/android-setup-steps-configs';
+
+export const configuringPortForwarding: AndroidSetupStepConfig = deps => ({
+    actions: {},
```
Comment (dbjorge) [inline src/electron/platform/android/setup/steps/configuring-port-forwarding.ts:7]: Should have a cancel handler to go back to `detect-devices`, probably

File: src/electron/platform/android/setup/steps/prompt-configuring-port-forwarding-failed.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AndroidSetupStepConfig } from 'electron/platform/android/setup/android-setup-steps-configs';
+
+export const promptConfiguringPortForwardingFailed: AndroidSetupStepConfig = deps => {
+    return {
+        actions: {
```
Comment (dbjorge) [inline src/electron/platform/android/setup/steps/prompt-configuring-port-forwarding-failed.ts:8]: Should have a cancel handler?

File: src/electron/platform/android/setup/android-setup-deps.ts [hunk 3]
```diff
@@ -11,4 +11,5 @@ export type AndroidSetupDeps = {
     hasExpectedServiceVersion: () => Promise<boolean>;
     installService: () => Promise<boolean>;
     hasExpectedPermissions: () => Promise<boolean>;
+    setTcpForwarding: () => Promise<boolean>;
```
Comment (RobGallo) [inline src/electron/platform/android/setup/android-setup-deps.ts:14]: Returns a bool; is this a question? Should this function name begin with 'has'?

File: src/electron/views/device-connect-view/components/android-setup/default-android-setup-components.ts [hunk 4]
```diff
@@ -26,5 +26,7 @@ export const defaultAndroidSetupComponents: AndroidSetupStepComponentProvider =
     'installing-service': InstallingServiceStep,
     'detect-devices': DetectDevicesStep,
     'detect-service': DetectServiceStep,
+    'configuring-port-forwarding': DetectAdbStep,
```
Comment (RobGallo) [inline src/electron/views/device-connect-view/components/android-setup/default-android-setup-components.ts:29]: These lines need to point to the correct steps. :)

File: src/tests/unit/tests/electron/platform/android/setup/steps/configuring-port-forwarding.test.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AndroidSetupStepConfigDeps } from 'electron/platform/android/setup/android-setup-steps-configs';
+import { configuringPortForwarding } from 'electron/platform/android/setup/steps/configuring-port-forwarding';
+import { Mock, MockBehavior, Times } from 'typemoq';
+import { checkExpectedActionsAreDefined } from './actions-tester';
+
+describe('Android setup step: configuringPortFowarding', () => {
```
Comment (RobGallo) [inline src/tests/unit/tests/electron/platform/android/setup/steps/configuring-port-forwarding.test.ts:9]:
```suggestion
describe('Android setup step: configuringPortForwarding', () => {
```

File: src/electron/platform/android/setup/steps/configuring-port-forwarding.ts (hunk 1 above)
Comment (dbjorge) [reply src/electron/platform/android/setup/steps/configuring-port-forwarding.ts:7]: Per offline discussion, leaving for a separate PR (other steps also missing them)

File: src/electron/platform/android/setup/steps/prompt-configuring-port-forwarding-failed.ts (hunk 2 above)
Comment (dbjorge) [reply src/electron/platform/android/setup/steps/prompt-configuring-port-forwarding-failed.ts:8]: Per offline discussion, leaving for a separate PR (other steps also missing them)

File: src/electron/platform/android/setup/android-setup-deps.ts (hunk 3 above)
Comment (DaveTryon) [reply src/electron/platform/android/setup/android-setup-deps.ts:14]: I'm continuing the pattern of not throwing from the deps. A false value = an error. I'll move the catch handler up into the onEnter

File: src/electron/views/device-connect-view/components/android-setup/default-android-setup-components.ts (hunk 4 above)
Comment (DaveTryon) [reply src/electron/views/device-connect-view/components/android-setup/default-android-setup-components.ts:29]: Placeholder values until Dan merges the steps

File: src/tests/unit/tests/electron/platform/android/setup/steps/configuring-port-forwarding.test.ts (hunk 5 above)
Comment (DaveTryon) [reply src/tests/unit/tests/electron/platform/android/setup/steps/configuring-port-forwarding.test.ts:9]: Fixed

File: src/electron/views/device-connect-view/components/android-setup/default-android-setup-components.ts (hunk 4 above)
Comment (DaveTryon) [reply src/electron/views/device-connect-view/components/android-setup/default-android-setup-components.ts]: These have real values now

**Pull request merged**